packages/-ember-data/tests/integration/request-state-service-test.ts

```diff
@@ -0,0 +1,229 @@
+      };
+
+      let { owner } = this;
+
+      let TestAdapter = EmberObject.extend({
+        findRecord() {
+          return Promise.resolve(personHash);
+        },
+        deleteRecord() {
+          return Promise.resolve();
+        },
+
+        updateRecord() {
+          return Promise.resolve();
+        },
+
+        createRecord() {
+          return Promise.resolve();
+        },
+      });
+
+      owner.register('adapter:application', TestAdapter, { singleton: false });
+
+      store = owner.lookup('service:store');
+
+      let promise = store.findRecord('person', '1');
+      let requestService = store.getRequestStateService();
+
+      // Relying on sequential lids until identifiers land
+      let identifier = identifierCacheFor(store).getOrCreateRecordIdentifier({ type: 'person', id: '1' });
+      let request = requestService.getPendingRequestsForRecord(identifier)[0];
+
+      assert.equal(request.state, 'pending', 'request is pending');
+      assert.equal(request.type, 'query', 'request is a query');
+      let requestOp = {
+        op: 'findRecord',
+        recordIdentifier: identifier,
+        options: {},
+      };
+      assert.deepEqual(request.request.data[0], requestOp, 'request op is correct');
+
+      let person = await promise;
+      let lastRequest = requestService.getLastRequestForRecord(identifier);
+      let requestStateResult = {
+        type: 'query' as const,
+        state: 'fulfilled' as const,
+        request: { data: [requestOp] },
+        response: { data: normalizedHash },
+      };
+      assert.deepEqual(lastRequest, requestStateResult, 'request is correct after fulfilling');
+      assert.deepEqual(
+        requestService.getPendingRequestsForRecord(identifier).length,
+        0,
+        'no pending requests remaining'
+      );
+
+      let savingPromise = person.save();
+      let savingRequest = requestService.getPendingRequestsForRecord(identifier)[0];
+
+      assert.equal(savingRequest.state, 'pending', 'request is pending');
+      assert.equal(savingRequest.type, 'mutation', 'request is a mutation');
+      let savingRequestOp = {
+        op: 'saveRecord',
+        recordIdentifier: identifier,
+        options: {},
+      };
+      assert.deepEqual(savingRequest.request.data[0], savingRequestOp, 'request op is correct');
+
+      await savingPromise;
+      let lastSavingRequest = requestService.getLastRequestForRecord(identifier);
+      let savingRequestStateResult = {
+        type: 'mutation' as const,
+        state: 'fulfilled' as const,
+        request: { data: [savingRequestOp] },
+        response: { data: undefined },
+      };
+      assert.deepEqual(lastSavingRequest, savingRequestStateResult, 'request is correct after fulfilling');
+      assert.deepEqual(
+        requestService.getPendingRequestsForRecord(identifier).length,
+        0,
+        'no pending requests remaining'
+      );
+    });
+
+    test('event subscription', async function(assert) {
```

Comment:
test descriptions should be a bit more declarative of what the scenario being tested is

Reply:
done